google_guest_agent/network/systemd_networkd.go

```diff
@@ -0,0 +1,112 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/run"
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+const configDir = "/etc/systemd/network"
+
+type systemdnetworkd struct{}
+
+func (n systemdnetworkd) IsManaging(ctx context.Context, iface string) (bool, error) {
+	// First check if the service is running.
+	res := run.WithOutput(ctx, "/bin/sh", "-c", "systemctl status systemd-networkd.service | grep Active:")
+	if res.ExitCode != 0 {
+		return false, fmt.Errorf("failed to get status of systemd-networkd service: %v", res.StdErr)
+	}
+	if !strings.Contains(res.StdOut, "running") {
+		// Service is not running, so it can't be managing an interface.
+		return false, nil
+	}
+
+	// Check systemd network configuration.
+	res = run.WithOutput(ctx, "/bin/sh", "-c", "networkctl status | grep "+iface)
+	if res.ExitCode != 0 {
+		return false, fmt.Errorf("failed to check systemd-networkd network status: %v", res.StdErr)
+	}
+	if strings.Contains(res.StdOut, "configured") {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (n systemdnetworkd) Name() string {
```

Comment:
Even that it's pretty obvious what this method is doing please consider adding comments/docs to it mentioning that it's the implementation of the interface etc.